src/transports/json.ts

```diff
@@ -0,0 +1,39 @@
+import fs from "fs/promises";
+import path from "path";
+import { LogEntry } from "../log-entry";
+import { Transport } from "./transport";
+
+export class JsonArrayTransport implements Transport {
+  private logDirectory: string;
+
+  constructor(logDirectory: string = "./jsonTransportData") {
```

Comment:
You can put them all in the "data" directory. There should be no conflicts given the files have different extensions.